### PR TITLE
add max entity size in VertxHttpExchange

### PR DESCRIPTION
**Pull request by machi1990**
Fixes https://github.com/quarkusio/quarkus/issues/3372

This is required for https://github.com/quarkusio/quarkus/pull/3404 to land.

I could not find a way to properly add tests for this change on this repo, so I based myself on the tests introduced on the cited PR above. Any pointers are appreciated. 

/cc @stuartwdouglas @gsmet

### CODE REVIEW

File: vertx/src/main/java/io/undertow/vertx/VertxHttpExchange.java [hunk 1]
```diff
@@ -509,6 +512,16 @@ public void handle(Buffer event) {
         BiConsumer<InputChannel, Object> readCallback = null;
         Object context = null;
         synchronized (request.connection()) {
+            uploadSize += event.length();
+            if (maxEntitySize != UndertowOptions.DEFAULT_MAX_ENTITY_SIZE && uploadSize > maxEntitySize) {
+                responseDone = true;
+                terminateRequest();
+                response.setStatusCode(413);
```
Comment (stuartwdouglas) [inline vertx/src/main/java/io/undertow/vertx/VertxHttpExchange.java]: This should probably check if a response has already been sent or not, and if so just call .close(). It also should set the Connection: close header.

Comment (machi1990) [reply vertx/src/main/java/io/undertow/vertx/VertxHttpExchange.java]: Thanks for the review @stuartwdouglas. I have addressed this part.

File: vertx/src/main/java/io/undertow/vertx/VertxHttpExchange.java [hunk 2]
```diff
@@ -349,14 +351,7 @@ public int readBytesAvailable() {
     public ByteBuf readBlocking() throws IOException {
         synchronized (request.connection()) {
             while (input1 == null && !eof) {
-                try {
```
Comment (stuartwdouglas) [inline vertx/src/main/java/io/undertow/vertx/VertxHttpExchange.java]: This is broken, there is different reading and writing versions with different state variabled (waitingForRead and waitingForWrite)

Comment (machi1990) [reply vertx/src/main/java/io/undertow/vertx/VertxHttpExchange.java]: Oops, my IDE proposed the change. I am reverting this change, it is not in the scope of the PR also. Thanks.

Comment (machi1990) [reply vertx/src/main/java/io/undertow/vertx/VertxHttpExchange.java]: Reverting done. Thanks for spotting this @stuartwdouglas , I should be more careful next time :-)

**Pull request merged**